### PR TITLE
Enable checkstyle

### PR DESCRIPTION
**Pull request by bmeike**
Not currently failing on errors

### CODE REVIEW

File: realm/realm-library/build.gradle [hunk 1]
```diff
@@ -236,23 +236,32 @@ task pmd(type: Pmd) {
...
+    toolVersion ="7.6"
 
     configFile = file("${projectDir}/../config/checkstyle/checkstyle.xml")
 
-    // empty classpath
-    classpath = files()
+    def configProps = ['proj.module.dir': projectDir.absolutePath]
+    configProperties configProps
+
+    ignoreFailures = true
```
Comment (beeender) [inline realm/realm-library/build.gradle]: Just curious, what does this do?

**Comment by beeender** [comment]
uh, right, please rebase this on `releases` branch instead. Just let me know if you need help for rebasing. It would be easy just do:
```
git pull
git checkout checkstyle
git branch tmp
git stash
git reset --hard origin/releases
git cherry-pick tmp
git push origin/checkstyle --force
```

then edit this PR, choose the base branch in the editing page.

File: realm/realm-library/build.gradle (hunk 1 above)
Comment (bmeike) [inline realm/realm-library/build.gradle]: Heh!  I think it just does this:`
configProperties configProps = ['proj.module.dir': projectDir.absolutePath]
`
I copied it over, exactly, from the pre-existing code, 'cause I didn't understand it, either!

**Comment by bmeike** [comment]
@beeender Hah!  I believe I can do it with 2 fewer commands:

`git checkout -b tmp
git fetch
git reset --hard origin/releases
git cherry-pick blake/checkstyle
git push -f origin HEAD:checkstyle
`

**Comment by bmeike** [comment]
@cmelchior It was, mostly, working.  There are 3 changes:

**Pull request merged**